src/prommis/solid_handling/crusher.py

```diff
@@ -0,0 +1,231 @@
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+r"""
+Crusher 
+=======
+
+Author: Lingyan Deng
+
+The Crusher module including power consumption for solid crushing. It is a function of particle size distribution, mass flow rate, and bond work index.
+
+Degrees of Freedom
+------------------
+
+A Crusher module has two degree of freedom, which are the output of "particle_size_median" and "particle_size_width". 
+
+Model Structure
+---------------
+
+The Crusher model includes one inlet Port (inlet) and one outlet Port (outlet). The properties of the Crusher Unit model is mainly the particle size distribution. 
+
+Additional Constraints
+----------------------
+
+Crusher add one additional constraint to calculate the work required to crush the particles.
+
+.. math:: work_{t} = 10 \times flow_mass_in_{t} \times bond_work_index \times \left(\frac{1}{\sqrt{prod\_p80}}_{t} - \frac{1}{\sqrt{feed\_p80}}_{t}\right) 
+
+where :math: `work_{t}` is the work required to crush the particles at `t` time, 10 is an empirical value and should not be changed, :math: `flow_mass_in_{t}` is the inlet mass flow rate
+at `t` time, :math: `bond_work_index` is the bond work index of particles, :math: `prod\_p80_{t}` is production particle size with 80% passing the mesh at `t` time, :math: `feed\_p80_{t}` is 
+feed particle size with 80% passing the mesh at `t` time.
+
+Expressions
+-----------
+
+Crusher includes two expressions to calculate the size of particles that has 80% passing the mesh for both feed and product particles.
+
+.. math:: feed_p80_{t} = \frac{particle_size_median_{in_{t}}}{sunit} \times \left(-\log(1 - 0.8)\right)^{\frac{particle_size_width_{in_{t}}}{2}}
+
+where :math: `feed_p80_{t}` is the feed particle size that has 80% passing the mesh at `t` time, :math: `particle_size_median_{in_{t}}}{sunit}` is the median partile size of input at `t` time and unitless. The 
+default particle size is in micrometer, the expression convert it to unitless to avoide unit error in the work_{t} constraint. The :math: `particle_size_width_{in_{t}}` is the partile size width of input at `t` time. 
+
+.. math:: prod_p80_{t} = \frac{particle_size_median_{out_{t}}}{sunit} \times \left(-\log(1 - 0.8)\right)^{\frac{particle_size_width_{out_{t}}}{2}}
+
+where :math: `prod_p80_{t}` is the product particle size that has 80% passing the mesh at `t` time, :math: `particle_size_median_{out_{t}}}{sunit}` is the median partile size of output at `t` time and unitless. The 
+default particle size is in micrometer, the expression convert it to unitless to avoide unit error in the work_{t} constraint. The :math: `particle_size_width_{out_{t}}` is the partile size width of output at `t` time. 
```

Comment:
```suggestion
default particle size is in micrometer, the expression converts it to unitless to avoid unit error in the work_{t} constraint. The :math: `particle_size_width_{out_{t}}` is the partile size width of output at `t` time. 
```